### ISSUE TITLE
CelebA dataset generates mismatched data and labels

### ISSUE THREAD

**Issue by ma0li** [pytorch/vision#4209]
## 🐛 Bug

From [line 102](https://github.com/pytorch/vision/blob/8d540228b8ac8b52749d32ecf5782aef45391775/torchvision/datasets/celeba.py#L102), it is clearly to see that without indexing with "mask", self.filename consists of all samples from train/test/valid sets. However, all the other meta-labels are sliced with indexes "mask". This definitely will mismatch the data and the label when sampling.

## To Reproduce

This can be easily verified by checking the length of self.filename and self.attr, which are supposed to be identical.

1. add "assert len(self.attr) == len(self.filename)" after line 109

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch / torchvision Version (e.g., 1.0 / 0.4.0):
 - OS (e.g., Linux):
 - How you installed PyTorch / torchvision (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:
